src/lib/utils/index.ts

```diff
@@ -162,6 +162,10 @@ export const isFulfilled = <T>(
 export const isRejected = <T>(promise: PromiseSettledResult<T>): promise is PromiseRejectedResult =>
   promise.status === 'rejected'
 
+export const isErrorPromise = <U, V, T extends Result<U, V>>(
+  promise: PromiseSettledResult<T>,
+): promise is PromiseRejectedResult => 'value' in promise && promise.value.isErr()
+
```

Comment:
❤️

Reply:
Two nitpicks naming-wise here:

- The arg isn't a promise but a settled promised result (i.e the return value of `Promise.allSettled()`), which are quite different
- `isErr()` is only present because the value is a Result monad 

I'd be more inclined to something fully declarative like `isResolvedErr` (i.e resolved promise, but unwrapping yields an error monad)

Reply:
Happy with your proposition, I pushed it!